Formula/crystal.rb

```diff
@@ -75,17 +80,27 @@ def install
 
     (buildpath/"boot").install resource("boot")
     ENV.append_path "PATH", "boot/bin"
-    ENV.append_path "CRYSTAL_LIBRARY_PATH", Formula["bdw-gc"].opt_lib
-    ENV.append_path "CRYSTAL_LIBRARY_PATH", ENV["HOMEBREW_LIBRARY_PATHS"]
-    ENV.append_path "CRYSTAL_LIBRARY_PATH", Formula["libevent"].opt_lib
+    ENV["CRYSTAL_LIBRARY_PATH"] = ENV["HOMEBREW_LIBRARY_PATHS"]
+    ENV.append_path "CRYSTAL_LIBRARY_PATH", MacOS.sdk_path_if_needed/"usr/lib" if MacOS.sdk_path_if_needed
     ENV.append_path "LLVM_CONFIG", llvm.opt_bin/"llvm-config"
 
+    crystal_install_dir = libexec
+    # Avoid embedding HOMEBREW_PREFIX references in `crystal` binary.
+    config_library_paths = ENV["CRYSTAL_LIBRARY_PATH"].gsub(
+      HOMEBREW_PREFIX,
+      "\\$$ORIGIN/#{HOMEBREW_PREFIX.relative_path_from(crystal_install_dir)}",
+    )
```

Comment:
I think there is no need for including all of `HOMEBREW_LIBRARY_PATHS` here. This could probably just be a single path to `$HOMEBRE_PREFIX/lib`, relative from the binary location.

This should be equivalent to the configuration that's currently in the wrapper script (https://github.com/Homebrew/homebrew-core/pull/105150/files#diff-16c9966e7b0364c7c4024aefcc341498cb8359003b6e85ba6212ddcd530bba22R136) and which effectively overrides this compile-time configuration.

Reply:
> I think there is no need for including all of `HOMEBREW_LIBRARY_PATHS` here. This could probably just be a single path to `$HOMEBRE_PREFIX/lib`, relative from the binary location.

I suggest keeping all/most of `HOMEBREW_LIBRARY_PATH`s, unless the extra paths really add so much extra overhead to `crystal`.

Using a `$HOMEBREW_PREFIX/opt/foo/lib` path hardens `crystal` against the user doing `brew unlink foo` or from `foo` become keg-only for some reason.

Reply:
Actually, having looked at what's in `HOMEBREW_LIBRARY_PATH`, it looks like the specific paths it specifies aren't useful for `crystal`, and we've got no complaints about `brew unlink` problems so far. It's probably safe to do as you suggest here, and I've opted for that.